src/main/java/com/javacode/dao/impl/UserDAOImpl.java

```diff
@@ -22,7 +22,7 @@ public UserDAOImpl() {
 	@Override
 	public User findByEmail(String email) {
 		logger.info("email: " + email);
-		return (User) getSession().createQuery("FROM User where email = :email").setParameter("email", email).getSingleResult();
+		return (User) getSession().createQuery("FROM User where email = :email").setParameter("email", email).getResultList().stream().findFirst().orElse(null);
```

Comment:
Sao ko dùng single result mà là result list rồi lại phải lấy record đầu tiên vất vả ri chú

Reply:
em dùng single lấy ra thì bị báo lỗi NoResultException No entity found for query nên em chuyển thành tnay ạ

Reply:
ukm, bjk rồi
nếu rứa ở service đoạn catch, em có thể return null là dc, ở service có bắt exception mà